### PR TITLE
Implementing metrics dynamic plugin

### LINKED ISSUE #25886: Metrics Stage 3: Save metrics data to Analytics Infrastructure
We need to connect the Experience plugin to the analytics infrastructure, so that when the plugin is installed on a system, the data it collects can be sent to the analytics DB.

### Data Format
* The metrics data will be sent to the analytics infrastructure as a JSON string.
    * Note that the format may change later, when a reporting tool is selected, to optimize the data analysis or reporting.  But for now we just want to ensure that the data can be sent successfully.

### Execution
* Sending of the metrics data should be executable via a Java method.
    * It should _not_ be triggerable via an API method.

### Logging
* DEBUG level logging should be implemented to log every time the data is sent.
    * This should include, at a minimum, the full data sent (the JSON string) and the response from the analytics infrastructure.
* WARN level logging should be implemented to log any time the data fails to send.
    * This should include, only the error state and error message.
    * It should not include the full data sent.
* No INFO level logging should be implemented.

### PR DESCRIPTION
**Pull request by victoralfaro-dotcms**
Introducing metrics plugin in existent com.dotcms.experience plugin with metrics sender broadcasting and exposing a metrics REST endpioint.

Refs: #25866

### CODE REVIEW

**Review by jcastro-dotcms — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/metrics/MetricsSender.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package com.dotcms.metrics;
+
+import com.dotcms.http.request.StringPayloadHttpRequest;
+
+public interface MetricsSender {
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/metrics/MetricsSender.java:5]: doc

File: dotCMS/src/main/java/com/dotcms/metrics/MetricsSender.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+    String TOKEN_QUERY_PARAM_NAME = "token";
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/metrics/MetricsSender.java:7]: what is that for?

File: dotCMS/src/main/resources/osgi/osgi-extra.conf [hunk 3]
```diff
@@ -572,6 +572,9 @@ com.dotmarketing.portlets.report.action,
 com.dotmarketing.common.reindex,
 com.dotmarketing.business.cache.provider.timedcache,
 com.dotmarketing.portlets.calendar.model,
+com.dotcms.metrics,
```
Comment (jdotcms) [inline dotCMS/src/main/resources/osgi/osgi-extra.conf:575]: good

File: dotCMS/src/main/java/com/dotcms/metrics/MetricsAPI.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
+package com.dotcms.metrics;
+
+import com.dotcms.analytics.app.AnalyticsApp;
+import com.dotcms.http.request.StringPayloadHttpRequest;
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/metrics/MetricsAPI.java:4]: I would keep anything related to http requests outside of the scope of the API.

File: dotCMS/src/main/java/com/dotcms/metrics/MetricsAPI.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+     * @param key, key to identify metrics, probably the sender implementation class name
+     */
+    void unsubscribeFromMetrics(final String key);
+
+    /**
+     * Sends metrics to analytics.
+     *
+     * @param httpRequest {@link StringPayloadHttpRequest} instance with metrics data
+     */
+    void sendMetrics(final StringPayloadHttpRequest httpRequest);
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/metrics/MetricsAPI.java:35]: Same comment from above applies.

File: dotCMS/src/main/java/com/dotcms/metrics/MetricsSender.java (hunk 1 above)
Comment (victoralfaro-dotcms) [reply dotCMS/src/main/java/com/dotcms/metrics/MetricsSender.java:5]: Added.

File: dotCMS/src/main/java/com/dotcms/metrics/MetricsSender.java (hunk 2 above)
Comment (victoralfaro-dotcms) [reply dotCMS/src/main/java/com/dotcms/metrics/MetricsSender.java:7]: Relocated to `MetricsAPI`

File: dotCMS/src/main/java/com/dotcms/metrics/MetricsAPI.java (hunk 4 above)
Comment (victoralfaro-dotcms) [reply dotCMS/src/main/java/com/dotcms/metrics/MetricsAPI.java]: Refactored.

File: dotCMS/src/main/java/com/dotcms/metrics/MetricsAPI.java (hunk 5 above)
Comment (victoralfaro-dotcms) [reply dotCMS/src/main/java/com/dotcms/metrics/MetricsAPI.java]: Refactored.

**Review by dsilvam — APPROVED** [review]

**Pull request merged**